internal/database/actions.go

```diff
@@ -38,6 +42,60 @@ func ListMeasurements(sess sqlbuilder.Database, resultID int64) ([]MeasurementUR
 	return measurements, nil
 }
 
+type MeasurementWithInput struct {
+	Input string `json:"input"`
+}
+
+func GetMeasurementJSON(sess sqlbuilder.Database, measurementID int64) (map[string]interface{}, error) {
+	var (
+		measurement MeasurementURLNetwork
+		msmtJSON map[string]interface{}
+	)
+
+	req := sess.Select(
+		db.Raw("urls.*"),
+		db.Raw("measurements.*"),
+	).From("measurements").
+		LeftJoin("urls").On("urls.url_id = measurements.url_id").
+		Where("measurements.measurement_id= ?", measurementID)
+
+	if err := req.One(&measurement); err != nil {
+		log.Errorf("failed to run query %s: %v", req.String(), err)
+		return nil, err
+	}
+	reportFilePath := measurement.Measurement.ReportFilePath
+	if (measurement.URL.URL.Valid == false) {
+		b, err := ioutil.ReadFile(reportFilePath)
+		if err != nil {
+			return nil, err
+		}
+		json.Unmarshal(b, &msmtJSON)
+		return msmtJSON, nil
+	}
+
+	url := measurement.URL.URL.String
```

Comment:
Oh, I am also confused by `.URL.URL`: why twice?

Reply:
It is twice because `URL`[0] is the `url` table, while `URL`[1] is the `url` column inside of the `url` table.

Reply:
In general, I find it bad to have a `.Foo.Foo` chaining where it's unclear to me what `Foo` and `Foo` are. At the same time, I believe the naming cannot change at this point, and changing it would be non-practical :-(.

Reply:
If you look at the `MeasurementURLNetwork` model inside of `internal/database/models.go` I think it will improve your understanding of what `URL` and `URL` are (one is the inline definition of the table, while the other is the attribute mapped to the column in the table).

I don't think it's that uncommon to have this sort of database schema where you have a column in the table named like the table itself.

Reply:
Sure, if I knew better the code I'd not be surprised. But that the code is structured in a way that it is a bit surprising is a bummer. I guess this is a side effect of using an ORM. Also, it may be that, when using a IDE, it's easier to know what is what. Instead, when doing a review only looking at a diff, this looks much worst. (But maybe this is a feature).

In any case, I don't think there's much we can do about that 🤷‍♂

Reply:
> I don't think it's that uncommon to have this sort of database schema where you have a column in the table named like the table itself.

I have seen `URLS.URL` and `URL.URL`. There are arguments for both designs. We have a design with `URL.URL` and we probably don't want to change it.